web/app_options.js

```diff
@@ -13,13 +13,110 @@
  * limitations under the License.
  */
 
+import { viewerCompatibilityParams } from './viewer_compatibility';
+
 const OptionKind = {
   VIEWER: 'viewer',
   API: 'api',
   WORKER: 'worker',
 };
 
 const defaultOptions = {
+  defaultZoomValue: {
+    /** @type {string} */
+    value: '',
+    kind: OptionKind.VIEWER,
+  },
+  disableFullscreen: {
+    /** @type {boolean} */
+    value: viewerCompatibilityParams.disableFullscreen || false,
+    kind: OptionKind.VIEWER,
+  },
+  disableHistory: {
+    /** @type {boolean} */
+    value: false,
+    kind: OptionKind.VIEWER,
+  },
+  disablePageLabels: {
+    /** @type {boolean} */
+    value: false,
+    kind: OptionKind.VIEWER,
+  },
+  disablePageMode: {
+    /** @type {boolean} */
+    value: false,
+    kind: OptionKind.VIEWER,
+  },
+  enablePrintAutoRotate: {
+    /** @type {boolean} */
+    value: false,
+    kind: OptionKind.VIEWER,
+  },
+  enableWebGL: {
+    /** @type {boolean} */
+    value: false,
+    kind: OptionKind.VIEWER,
+  },
+  externalLinkRel: {
+    /** @type {string} */
+    value: null,
```

Comment:
> Moreover, since this is a string type, I would at least expect `''`.

You're absolutely correct that this should be a string, however an empty string wouldn't work here; see: https://github.com/mozilla/pdf.js/blob/5f98f9b8f10d68df6cbd331b5e135e167c71829b/src/display/dom_utils.js#L317

I've changed this particular line to explicitly list the "default" link rel instead; but see below as well.

> If this is to avoid a dependecy on code from src, can we add a comment somewhere in this file to indicate that?

Correct, I view this (new) file somewhat similar to https://github.com/mozilla/pdf.js/blob/master/web/default_preferences.json. Even though we can, as opposed to a `.json` file, import constants/types from elsewhere here I nonetheless wanted to avoid that pattern. I've tried to add a comment to that effect.